src/main/java/uk/gov/ida/matchingserviceadapter/saml/UserIdHashFactory.java

```
@@ -4,47 +4,51 @@
 import java.security.NoSuchAlgorithmException;
 import java.text.MessageFormat;
 
-import static com.google.common.base.Throwables.propagate;
-
 public class UserIdHashFactory {
-    public String createHashedId(String partnerEntityId, String entityId, String persistentId, Optional<IdentityProviderAuthnStatement> authnStatement) {
+
+    private final String msaEntityId;
+
+    public UserIdHashFactory(String msaEntityId) {
+        this.msaEntityId = msaEntityId;
+    }
+
+    public String hashId(String issuerEntityId, String persistentId, Optional<AuthnContext> authnContext) {
         MessageDigest md;
         try {
             md = MessageDigest.getInstance(JCAConstants.DIGEST_SHA256);
         } catch (NoSuchAlgorithmException e) {
-            throw propagate(e);
+            throw new RuntimeException(e);
         }
 
-        final String persistentIdHash = getPersistentIdHashForDigest(partnerEntityId, entityId, persistentId, authnStatement);
+        final String toHash = idToHash(issuerEntityId, persistentId, authnContext);
 
         try {
-            md.update(persistentIdHash.getBytes("UTF-8"));
+            md.update(toHash.getBytes("UTF-8"));
         } catch (UnsupportedEncodingException e) {
-            throw propagate(e);
+            throw new RuntimeException(e);
         }
 
         byte[] digest = md.digest();
         return Hex.encodeHexString(digest);
     }
 
-    private String getPersistentIdHashForDigest(String partnerEntityId, String entityId, String persistentId, Optional<IdentityProviderAuthnStatement> authnStatement) {
+    private String idToHash(String issuerEntityId, String persistentId, Optional<AuthnContext> context) {
         String persistentIdHash;
 
-        final AuthnContext authnContext = authnStatement.get().getAuthnContext();
+        final AuthnContext authnContext = context.toJavaUtil().orElseThrow(() -> new IllegalStateException(String.format("Authn context absent for persistent id %s", persistentId)));
```

Should throw the custom exception, not IllegalStateException...